### PR TITLE
fix(fromgeometry) : Add hole curves along with boundary curve

### PR DESCRIPTION
**Pull request by devngc**
HI @chriswmackey,

HB Color Face Attributes component only outputs the curve for the boundary. This PR fixes it so that the component outputs curves for hole as well when a face has holes.

This will require `extend` instead of `append`  on [this](https://github.com/ladybug-tools/honeybee-grasshopper-core/blob/master/honeybee_grasshopper_core/src/HB%20Color%20Face%20Attributes.py#L90) line.

### CODE REVIEW

File: ladybug_rhino/fromgeometry.py [hunk 1]
```diff
@@ -141,8 +141,17 @@ def from_polyface3d(polyface):
 
 def from_face3d_to_wireframe(face):
     """Rhino PolyLineCurve from ladybug Face3D."""
-    return rg.PolylineCurve(
-        [from_point3d(pt) for pt in face.boundary] + [from_point3d(face.boundary[0])])
+    if face.has_holes == True:
```
Comment (chriswmackey) [inline ladybug_rhino/fromgeometry.py:144]: The PEP8 way to do it is to just say `if face.has_holes:`.

Here's the [official recommendation](https://www.python.org/dev/peps/pep-0008/#programming-recommendations):
```
Yes:   if greeting:
No:    if greeting == True:
Worse: if greeting is True:
```

File: ladybug_rhino/fromgeometry.py [hunk 2]
```diff
@@ -141,8 +141,17 @@ def from_polyface3d(polyface):
...
+        hole_curves = [rg.PolylineCurve(
+            [from_point3d(pt) for pt in tup] + [from_point3d(tup[0])]) for tup in face.holes]
+        boundary_curve = [rg.PolylineCurve(
+            [from_point3d(pt) for pt in face.boundary] + [from_point3d(face.boundary[0])])]
+        curves = boundary_curve + hole_curves
+        return curves
+    else:
+        boundary_curve = [rg.PolylineCurve(
+            [from_point3d(pt) for pt in face.boundary] + [from_point3d(face.boundary[0])])]
+        return boundary_curve
```
Comment (chriswmackey) [inline ladybug_rhino/fromgeometry.py:154]: I think we need to be consistent now and always return a list. So can you change this to be a list with one item and also change the docstring of this method accordingly.

**Review by chriswmackey — COMMENTED** [review]
Thank you very much for finding and fixing this bug, @devngc .  I just have two styling comments and then we can merge.

**Comment by chriswmackey** [comment]
FYI, changing this method to output a list is going to break all of the [honeybee-grasshopper-core 1:: Visualize components](https://github.com/ladybug-tools/honeybee-grasshopper-core) with a wireframe output. Do you think you could send a corresponding PR that changes the `append` inside these components to `extend`?

File: ladybug_rhino/fromgeometry.py (hunk 2 above)
Comment (devngc) [reply ladybug_rhino/fromgeometry.py:154]: @chriswmackey, I beleive, it is a list with one item. That is the curve for the boundary.

**Comment by chriswmackey** [comment]
Ah. And we will also have to change the honeybee-grasshopper-energy "Color Rooms" and "Color Faces" components.

**Comment by chriswmackey** [comment]
.. and the "DF Visualize Wireframe" component.

**Comment by devngc** [comment]
Hi @chriswmackey,

I created PR for following components

HB Color Face Attributes, HB Label Faces, Hb Visualize by Type, HB Visualize by BC,
https://github.com/ladybug-tools/honeybee-grasshopper-core/pull/160

HB Color Faces
https://github.com/ladybug-tools/honeybee-grasshopper-energy/pull/161

DF Vizualize Wireframe
https://github.com/ladybug-tools/dragonfly-grasshopper/pull/95

File: ladybug_rhino/fromgeometry.py (hunk 1 above)
Comment (devngc) [reply ladybug_rhino/fromgeometry.py]: Complied

**Comment by devngc** [comment]
[Here](https://github.com/devngc/Work-in-Progress/blob/master/reference/test.gh) is a link to a Gh file with the edited components if you would like to test.

**Review by chriswmackey — APPROVED** [review]
Thanks @devngc .
This one looks good. Once you address the requested changes on the two honeybee-grasshopper PRs, we can merge.

**Pull request merged**